### DISCUSSION TITLE
How to read to data from smart contract without having to connect the wallet

### DISCUSSION THREAD

**Discussion by Bulalu** [wevm/wagmi#1994] [Q&A]
I want users who visit my to not have the need to connect their in order for them to some info from the smart contract, example totalSupply, price and etc.

I'm using two providers atm which are alchemy and the public provider, if i'm not mistaking i had this working a couple days back but now  i keep a 429 error, from this url (https://eth-mainnet.alchemyapi.io/v2/_gg7wSSi0KMBsdKnGVfHDueq6xMB9EkC 429)

**Comment by justingreenberg** [comment]
429 means you are being rate limited by rpc provider

you may need to provide alchemy key or upgrade your plan or if you are using free tier add an infura key as fallback

**Reply by Bulalu** [reply]
thanks for the reply justin, i did that but now i get a different error 

`index.js:1 Error: call revert exception [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION ] (method="symbol()", data="0x", errorArgs=null, errorName=null, errorSignature=null, reason=null, code=CALL_EXCEPTION, version=abi/5.7.0)
    at Logger.makeError (index.ts:269:1)
    at Logger.throwError (index.ts:281:1)
    at Interface.decodeFunctionResult (interface.ts:427:1)
    at _callee6$ (index.ts:400:1)
    at tryCatch (regeneratorRuntime.js:44:1)
    at Generator.<anonymous> (regeneratorRuntime.js:125:1)
    at Generator.next (regeneratorRuntime.js:69:1)
    at fulfilled (index.ts:1:1)`
    
   

 one wierd thing is that i connect my wallet this error disappears, and it all works

>

**Comment by SaadNiaziDev** [comment]
Can u share your code snippet? It would be helpful
